### ISSUE TITLE
No way to disable `ESC` key to close the modal

### ISSUE THREAD

**Issue by sep2** [mantinedev/mantine#588]
### What package has an issue

@mantine/modals

### Describe the bug

In the code below, although `onClose` is set to an empty function, the opened modal is still closed when user press `ESC` key in the keyboard.

```ts
    modals.openContextModal(modalId, {
        closeOnClickOutside: false,
        onClose: () => void 0,
    })
```

### In which browser did the problem occur

Chrome latest version

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Allow the `ESC` key to close the modal to be disabled.

**Comment by rtivital** [comment]
Thanks for reporting, the feature to disable close on escape will be added in next minor release

**Comment by Ultra-Instinct-05** [comment]
The `Drawer` component already has the `noCloseOnEscape` prop, which closes the the drawer when <kbd>escape</kbd> is pressed. So the relevant logic https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/components/Drawer/Drawer.tsx#L136-L149 can be applied to `Modal` as well. Perhaps it would also be better to extract the logic above into a utility function that can then be shared among any component that needs to close on <kbd>escape</kbd>

**Comment by rtivital** [comment]
It's one if statement, no need to extract anything

**Comment by rtivital** [comment]
Fixed in 3.5.0, you can disable close on escape by setting `<Modal closeOnEscape={false} />`
